tx pool: Avoid passing redundant hash into addUnchecked (scripted-diff)

maflcko opened the pull request:
Several years ago the transaction hash was not cached. For optimization the hash was instead passed into `addUnchecked` to avoid re-calculating it. See f77654a0e9424f13cad04f82c014abd78fbb5e38

Passing in the hash is now redundant and the argument can safely be removed.

Commit:
Mark CTxMemPoolEntry members that should not be modified const

Commit:
tx pool: Use the entry's hash instead of the one passed to addUnchecked

fanquake commented:
Looks like the scripted-diff is failing because of the other whitespace/formatting changes also included in the commit

Commit:
scripted-diff: Remove unused first argument to addUnchecked

-BEGIN VERIFY SCRIPT-
git grep -l addUnchecked | xargs sed --regexp-extended -i -e 's/addUnchecked\([^)][^,]+,\s*/addUnchecked(/g'
-END VERIFY SCRIPT-

MarcoFalke commented:
This is split off of #13804, but not required. So if people hate this kind of refactoring, I am happy to drop it.

promag commented:
utACK fa58777. I like the change, passing entry is now enough. Could update OP now that #13786 is merged.

practicalswift commented:
Concept ACK

Empact commented:
utACK fa58777

laanwj commented:
utACK fe5c49766c0dc5beaf186d77b568361242b20d5e
removing the redundancy is good because it avoids a potential bug when passing in the wrong hash
